haproxy/tests/conftest.py

```diff
@@ -5,25 +5,89 @@
 import time
 import logging
 import mock
-import tempfile
-import shutil
 import getpass
 
-from . import common
+from copy import deepcopy
+
+from datadog_checks.dev import TempDir, WaitFor, docker_run
+from datadog_checks.dev.utils import create_file, file_exists
+from datadog_checks.haproxy import HAProxy
+
+from .common import HERE, CHECK_CONFIG, USERNAME, PASSWORD, STATS_URL, STATS_URL_OPEN
 
 log = logging.getLogger('test_haproxy')
 
 
+def wait_for_haproxy():
+    res = None
+    res_open = None
+    try:
+        auth = (USERNAME, PASSWORD)
+        res = requests.get(STATS_URL, auth=auth)
+        res.raise_for_status()
+        res_open = requests.get(STATS_URL_OPEN)
+        res_open.raise_for_status()
+        return
+    except Exception as e:
+        log.info("exception: {0} res: {1} res_open: {2}".format(e, res, res_open))
+
+    return False
```

Comment:
This whole function can just be:

```python
def wait_for_haproxy():
    res = requests.get(STATS_URL, auth=(USERNAME, PASSWORD))
    res.raise_for_status()

    res_open = requests.get(STATS_URL_OPEN)
    res_open.raise_for_status()
```

since logic of WaitFor already does the handling